lang/csharp/src/apache/test/Schema/SchemaTests.cs

```diff
@@ -261,6 +261,27 @@ public void TestArray(string s, string item)
             testToString(sc);
         }
 
+        [TestCase("{\"type\": \"int\", \"logicalType\": \"date\"}", "int", "date")]
+        public void TestLogicalPrimitive(string s, string baseType, string logicalType)
```

Comment:
Could you add test cases for the other supported logical types in `SchemaTests`?

Reply:
I was just referencing this branch with the Confluent Kafka Avro serdes to test it out, and was getting the below error for the below schema. Just quickly pasting this before disappearing for the weekend, haven't looked closely, will investigate further later. Perhaps some tests of more complex types would be worthwhile?

```
{"namespace": "Confluent.Kafka.Examples.AvroSpecific",
 "type": "record",
 "name": "User",
 "fields": [
     {"name": "favorite_number", "type": {"type": "bytes", "logicalType": "decimal", "precision": 4, "scale": 2 } },
 ]
}
```

```
   at System.String.Insert(Int32 startIndex, String value)
   at Avro.AvroDecimal.ToString() in /git/avro/lang/csharp/src/apache/main/AvroDecimal.cs:line 155
   at Avro.AvroDecimal.ToString(String format, IFormatProvider formatProvider) in /git/avro/lang/csharp/src/apache/main/AvroDecimal.cs:line 843
   at System.Text.StringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.FormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.Format(String format, Object arg0, Object arg1)
   at Confluent.Kafka.Examples.AvroSpecific.Program.<>c__DisplayClass0_0.<Main>b__0() in /git/confluent-kafka-dotnet/examples/AvroSpecific/Program.cs:line 101
```